src/gt4py/storage/storage.py

```diff
@@ -399,6 +399,8 @@ def _construct(cls, backend, dtype, default_origin, shape, alignment, layout_map
 
     @property
     def data(self):
+        if cp is not None:
```

Comment:
Why is this needed? I think this violates the convention that `storage.data` is returning the actual cupy/numpy buffer.

Reply:
This may be a problem on our side, the `fv3ser` test suite does not distinguish between numpy and cupy arrays. Could we add a convenience method to `storage`, such as `asnumpy`, that converts any storage to a numpy array?

Reply:
you can always do
* `storage.view(np.ndarray)` since currently it's a subclass of np.ndarray
* `np.asarray(storage)`
note though that both of these options currently give you the cpu buffer as is, it does not trigger synchronization. 
you could do `storage.device_to_host()` before any of these options to make sure that you get an up-to-date cpu buffer. (it may later go out of sync again, so you shouldn't keep a reference longer than you are sure nothing is written in the storage. 

`device_to_host` actually also works for cpu storages, just does nothing there. 
we could in principle add that `np.asarray(storage)` triggers a synchronization.

Reply:
Thank you @gronerl, `storage.device_to_host(force=True)` followed by `np.asarray(storage)` worked.

Reply:
is the force necessary? this is potentially unsafe if the host buffer happens to be up to date already.

Reply:
Yes, the code did not work without it, because `_is_device_modified` was not `True`. This may be an issue with the Dawn backends, I will look into it.

Reply:
Found the issue, and removed the `force`, thank you for the assistance.